pkg/sentry/kernel/fd_table.go

```diff
@@ -370,19 +409,14 @@ func (f *FDTable) NewFDsVFS2(ctx context.Context, fd int32, files []*vfs.FileDes
 		return nil, unix.EMFILE
 	}
 
-	if fd == f.next {
-		// Update next search start position.
-		f.next = fds[len(fds)-1] + 1
-	}
-
 	f.mu.Unlock()
 	return fds, nil
 }
 
-// NewFDVFS2 allocates a file descriptor greater than or equal to minfd for
+// NewFDVFS2 allocates a file descriptor greater than or equal to minFD for
 // the given file description. If it succeeds, it takes a reference on file.
-func (f *FDTable) NewFDVFS2(ctx context.Context, minfd int32, file *vfs.FileDescription, flags FDFlags) (int32, error) {
-	if minfd < 0 {
+func (f *FDTable) NewFDVFS2(ctx context.Context, minFD int32, file *vfs.FileDescription, flags FDFlags) (int32, error) {
```

Comment:
Hmm can't this just call `FDTable.NewFDsVFS2` with `files = []*vfs.FileDescription{file}`? The logic seems identical in both functions.

The point is that we should refactor this in some way to avoid duplicate logic.

Reply:
> Hmm can't this just call `FDTable.NewFDsVFS2` with `files = []*vfs.FileDescription{file}`? The logic seems identical in both functions.
> 
> The point is that we should refactor this in some way to avoid duplicate logic.

 I think you are right. `NewFDsVFS2` can fulfill what `NewFDVFS2` can do.
Is it better to open a new PR for it?

Reply:
You can just update it here IMO.

Reply:
ok